skylib/zip.bzl

```diff
@@ -60,22 +82,38 @@ def gunzip(ctx, artifact):
     """
     out = ctx.actions.declare_file(artifact.basename + ".nogz")
     toolchain_info = ctx.toolchains["@io_bazel_rules_docker//toolchains/docker:toolchain_type"].info
-    input_manifests = []
-    tools = []
-    gzip_path = toolchain_info.gzip_path
-    if toolchain_info.gzip_target:
-        gzip_path = toolchain_info.gzip_target.files_to_run.executable.path
-        tools, _, input_manifests = ctx.resolve_command(tools = [toolchain_info.gzip_target])
-    elif toolchain_info.gzip_path == "":
-        fail("gzip could not be found. Make sure it is in the path or set it " +
-             "explicitly in the docker_toolchain_configure")
-    ctx.actions.run_shell(
-        command = "%s -d < %s > %s" % (gzip_path, artifact.path, out.path),
-        input_manifests = input_manifests,
-        inputs = [artifact],
-        outputs = [out],
-        use_default_shell_env = True,
-        mnemonic = "GUNZIP",
-        tools = tools,
-    )
+    gzip_path = _gzip_path(toolchain_info)
+
+    if gzip_path == "":
```

Comment:
Given this if/else condition has a few subtleties, I suggest refactoring it into a function and re-using it for both gzip & gunzip and pass in parameters that change (I think there should only be 2-3 such parameters?)

Regarding the condition itself, IIUC what we want to do is the following in order of precedence:
1. If neither gzip_path nor gzip_target is specified, use internal tool
2. else if gzip_path is specified, use the gzip tool at the specified absolute path
3. else if gzip_target is specified, build the specified target and use it assuming it accepts the same flags as gzip

There's a possibility `if gzip_path == ""` at the beginning might break this because, if gzip_target is specified, gzip_path might be blank (or maybe it's None?). If gzip_path defaults to None instead of "" when gzip_target is set, your logic still works but given this is very subtle, I suggest commenting exactly which scenario we're hitting above the `if` and `else` conditions and the values we expect for the gzip_* variables.

Reply:
Done. Note that the gzip_path == "" was done _after_ resolving gzip_path from the gzip_path/gzip_target attributes.
I replaced it with `if not gzip_path` to avoid the None issue.
Also added some comments.